### PR TITLE
Do not check hasShape(*input_type) in getInputShape

### PR DESCRIPTION
**Pull request by liqunfu**
### Description
do not check hasShape(*input_type) in getInputShape. This is to work with input as a scalar cases.

### CODE REVIEW

**Comment by xadupre** [comment]
Maybe it is worth add a unit test since it was not caught by the existing ones.

**Comment by gramalingam** [comment]
Hmmm ... this is a significant change. I am not sure I understand why we need this. I think the implementation should handle scalars fine because we are checking for the presence of a `TensorShapeProto`, we are not checking the rank (which is based on the repeated `dim` field inside a `TensorShapeProto`). 

Do you have an example model that causes a problem?

More info about the proto encoding here: https://protobuf.dev/programming-guides/field_presence/ ... I think what we are doing should work fine with both proto2 and proto3.

**Comment by gramalingam** [comment]
@liqunfu : is this still a concern? Or, can we close this?

File: onnx/defs/shape_inference.h [hunk 1]
```diff
@@ -340,9 +340,9 @@ inline const TensorShapeProto& getInputShape(const InferenceContext& ctx, size_t
   if (value_case != TypeProto::kTensorType && value_case != TypeProto::kSparseTensorType) {
     fail_type_inference("Input ", n, "expected to be a tensor or a sparse tensor type in ", ctx.getDisplayName(), ".");
   }
-  if (!hasShape(*input_type)) {
-    fail_shape_inference("Input ", n, " must have a non null shape in ", ctx.getDisplayName(), ".");
-  }
+
+  // do not check hasShape(*input_type) here, as scalar input is valid and it will return a default empty
```
Comment (xadupre) [inline onnx/defs/shape_inference.h:344]: I'm wondering why this was not caught sooner. Do you have an example where it is needed? My guess is an error should be raised sooner. This change is not aligned with hasInputShape which requires hasShape to be true.

**Comment by gramalingam** [comment]
Closing this. Please feel free to reopen if this is still needed.

**Pull request closed without merging**